tests/mir-opt/const_array_locals.rs

```diff
@@ -0,0 +1,46 @@
+//@ test-mir-pass: GVN
+#![feature(repr_simd)]
+
+// #[repr(simd)]
+// struct F32x8([f32; 8]);
+
+// EMIT_MIR const_array_locals.main.GVN.diff
+// CHECK-LABEL: fn main(
+#[rustfmt::skip]
+pub fn main() {
+    // CHECK: let [[array_lit:_.*]]: [i32; 32];
+    // CHECK:     debug _arr => [[arr:_.*]];
+
+    let _arr = [
```

Comment:
With commits[3], this now returns as a immutable memplace: `Some(OpTy { op: Indirect(MemPlace { ptr: alloc3<imm>, meta: None, misaligned: None }), ty: [i32; 32] })`